flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartitionView.java

```diff
@@ -156,7 +156,8 @@ public BufferAndBacklog getNextBuffer() throws IOException, InterruptedException
 				checkState(nextBuffer.isFinished(),
 					"We can only read from SpillableSubpartition after it was finished");
 
-				newBacklog = parent.decreaseBuffersInBacklogUnsafe(nextBuffer.isBuffer());
+				parent.decreaseBuffersInBacklog(nextBuffer.isBuffer());
+				newBacklog = parent.getBuffersInBacklog();
```

Comment:
1. why have you split this method into two? What was wrong with `decreaseBuffersInBacklog` returning the backlog?
2. why are you synchronizing here second time? You should be able to use `getBuffersInBacklogUnsafe()`

Reply:
1. The current backlog is based on `flushTriggered` or `finished` status after changing, if we want to return backlog during `decreaseBuffersInBacklog`, we might need to pass these info into `decreaseBuffersInBacklog(boolean `flushTriggered` | `finished`)`. In order not to dirty this method, I split it into a separate method. Maybe it does not seem so bad if passing parameter into `decreaseBuffersInBacklog `. So we could still merge these two into one method if you prefer.

2. Yes, you are right. Here we could get backlog in unsafe way, no need to add the synchronized overhead.

Reply:
1. The current backlog is based on `flushTriggered` or `isFinished` status after changing, if we want to return backlog during `decreaseBuffersInBacklog`, we might need to pass these info into `decreaseBuffersInBacklog(boolean flushTriggered | isFinished)`. In order not to dirty this method, I split it into a separate method. Maybe it does not seem so bad if passing parameter into `decreaseBuffersInBacklog`. So we could still merge these two into one method if you prefer.

2. Yes, you are right. Here we could get backlog in unsafe way, no need to add the synchronized overhead.

Reply:
1. Ok, I get it now :) Thanks for pointing this out.

I think that adding this extra parameter to the `decreaseBuffersInBacklog` would indeed be a worse solution and it's better to just use `getBuffersInBacklogUnsafe` here.